fenix/app/src/main/java/org/mozilla/fenix/components/metrics/InstallReferrerMetricsService.kt

```diff
@@ -0,0 +1,187 @@
+                    referrerClient = null
+                }
+            },
+        )
+    }
+
+    override fun stop() {
+        referrerClient?.endConnection()
+        referrerClient = null
+    }
+
+    override fun track(event: Event) = Unit
+
+    override fun shouldTrack(event: Event): Boolean = false
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun recordInstallReferrer(settings: Settings, url: String?) {
+        val params = url?.let(UTMParams::fromUrl)
+        if (params?.isEmpty() != false) {
+            return
+        }
+        params.intoSettings(settings)
+
+        params.apply {
+            source?.let {
+                PlayStoreAttribution.source.set(it)
+            }
+            medium?.let {
+                PlayStoreAttribution.medium.set(it)
+            }
+            campaign?.let {
+                PlayStoreAttribution.campaign.set(it)
+            }
+            content?.let {
+                PlayStoreAttribution.content.set(it)
+            }
+            term?.let {
+                PlayStoreAttribution.term.set(it)
+            }
+        }
+    }
+}
+
+/**
+ * Descriptions of utm parameters comes from
+ * https://support.google.com/analytics/answer/1033863
+ * - utm_source
+ *  Identify the advertiser, site, publication, etc.
+ *  that is sending traffic to your property, for example: google, newsletter4, billboard.
+ * - utm_medium
+ *  The advertising or marketing medium, for example: cpc, banner, email newsletter.
+ * utm_campaign
+ *  The individual campaign name, slogan, promo code, etc. for a product.
+ * - utm_term
+ *  Identify paid search keywords.
+ *  If you're manually tagging paid keyword campaigns, you should also use
+ *  utm_term to specify the keyword.
+ * - utm_content
+ *  Used to differentiate similar content, or links within the same ad.
+ *  For example, if you have two call-to-action links within the same email message,
+ *  you can use utm_content and set different values for each so you can tell
+ *  which version is more effective.
+ */
+data class UTMParams(
+    val source: String?,
+    val medium: String?,
+    val campaign: String?,
+    val term: String?,
+    val content: String?,
+) {
+    companion object {
+        /**
+         * Derive a set of UTM parameters from a string URL.
+         */
+        fun fromUrl(url: String): UTMParams {
+            val sanitizer = UrlQuerySanitizer()
+            sanitizer.allowUnregisteredParamaters = true
+            sanitizer.unregisteredParameterValueSanitizer = UrlQuerySanitizer.getUrlAndSpaceLegal()
+            sanitizer.parseUrl(url)
+            return UTMParams(
+                source = sanitizer.getValue("utm_source"),
+                medium = sanitizer.getValue("utm_medium"),
+                campaign = sanitizer.getValue("utm_campaign"),
+                term = sanitizer.getValue("utm_term"),
+                content = sanitizer.getValue("utm_content"),
```

Comment:
I wonder if we should make these constants within the companion object - which could then be accessed by CustomAttributeProvider